front/lib/api/wrappers.ts

```diff
@@ -152,19 +157,80 @@ export function withPublicAPIAuthentication<T, U extends boolean>(
     keyAuth: U extends true ? Authenticator : null
   ) => Promise<void> | void,
   opts: {
+    isWorkspaceRoute?: boolean;
+    forceAuthWithToken?: boolean;
     isStreaming?: boolean;
     allowUserOutsideCurrentWorkspace?: U;
-  } = {}
+  } = { isWorkspaceRoute: true, forceAuthWithToken: false }
 ) {
-  const { allowUserOutsideCurrentWorkspace, isStreaming } = opts;
+  const {
+    allowUserOutsideCurrentWorkspace,
+    isStreaming,
+    isWorkspaceRoute,
```

Comment:
I'm not a big fan of this one. So far we have reserved `Authenticator` to endpoints that are below a specific `worksapceId`.

For all other endpoints that are user specific we would simply rely on `getUserFromSession`

I would be in favor of keeping Authenticator to be a workspace notion only and introduce here `getUserFromAuth0Token` that we could use for the /me endpoint.

Reply:
Agreed, removed!